src/Kernel32.Shared/Win32Exception.cs

```diff
@@ -46,7 +46,7 @@ public Win32Exception()
         /// <param name="error">The Win32 error code associated with this exception.</param>
         public Win32Exception(Win32ErrorCode error)
 #if DESKTOP
-            : base((int)error)
+            : this(error, error.GetMessage() ?? $"Unknown error (0x{(int)error:x8})")
```

Comment:
When you say "the OS returns the current language" what is the "current language" Do you mean the language that the OS itself is using to present its UI to you, or the language currently set on the thread?

Regardless, I think we should revert this last commit and go with the .NET Framework behavior. If that means the test fails for your configuration, my hope would be that it passes when the OS is configured to use the same language that the thread is set to. Do you think that would be the outcome?